numpy/random/mtrand.pyx

```diff
@@ -3639,6 +3640,17 @@ cdef class RandomState:
         tol : float, optional
             Tolerance when checking the singular values in covariance matrix.
             cov is cast to double before the check.
+        mode : { 'default', 'legacy', 'fast', 'factor'}, optional
```

Comment:
It would be much more informative to name the the arguments after the algorithm they use. 

I particularly disagree with the pejorative *`legacy`* - the svd is the most robust method to use, especially in the presence of near-singularity which would otherwise blow up other methods. It's only downfall is that it is slow - something which can be discussed in the description.

I would instead use:
```python
        mode : { 'eigh', 'svd', 'cholesky', 'factor'}, optional
```

Reply:
Agreed, although I don't think `factor` belongs there at all